deploy/charts/cert-manager/templates/deployment.yaml

```diff
@@ -163,6 +163,18 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
+          {{- if (.Values.resources.limits).cpu }}
```

Comment:
Out of curiosity, what are the use-cases for using a CPU limit in the case of the cainjector, controller, and webhook? In https://home.robusta.dev/blog/stop-using-cpu-limits, the general recommendation is to not use CPU limits unless there is a specific need. The three only possible situations where CPU limits make sense are:

1. To avoid reliance on unpredictable spare CPU cycles due to low CPU requests → I don't think this applies to any of the cert-manager components.
2. To get more predictable QoS → Unless mistaken, QoS isn't a strong concern for cert-manager components IMO.
3. To be being able to charge back on multi-tenant environment. For example, IBM's licensing relies on CPU limits:

   > The CPU limits of Pods are considered and capped at the capacity of the Worker Node(s). ([IBM_Licensing_Tools.pdf](https://www.ibm.com/about/software-licensing/assets/guides_pdf/IBM_Licensing_Tools.pdf))

I think (3) is the only use-case for using CPU limits. Is that why you need them?

Reply:
Personally I don't use CPU limits at all; the blog above contains some fine explanation why you probably shouldn't. This PR does not in any way attempt to promote setting CPU limits; I'd still advise against it for almost all use cases.

However, if one does decide to set CPU limits for some reason, I think setting the value `GOMAXPROCS` to match with it will *generally* result in better performance (see e.g. https://github.com/uber-go/automaxprocs), so this PR is beneficial in that regard.

On the other hand, I'm also perfectly fine with dropping the `GOMAXPROCS` part as I personally don't intend to use it.

Reply:
Thanks for the precisions. I think it is still good to keep the `GOMAXPROCS` part for anyone forced to use CPU limits (such as with IBM Kubernetes); thank you for doing that!

Unrelated, but at some point I'd like to add a comment in `values.yaml` to advise people not to use `limits.cpu` unless they are forced to.